cmd/ingest/source/source.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Malicious Packages Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	ErrInvalidID  = errors.New("invalid source id")
+	validIDRegExp = regexp.MustCompile("[a-z0-9-]+")
+)
+
+type Source struct {
+	ID              string `yaml:"id"`
+	Bucket          string `yaml:"bucket"`
+	Prefix          string `yaml:"prefix"`
+	LookbackEntries int    `yaml:"lookback-entries"`
```

Comment:
Could you explain why this needs to be configurable rather than a constant?

Reply:
That's a fair point. It could be a constant for sure.

I was hoping to eventually have an end-to-end test for the tool to ensure it is working correctly so an external config would be useful.

But yes, it could be hard-coded.

Reply:
I might leave it as is. I do find it useful for local development.

Reply:
ack, fair enough.